zerver/lib/bugdown/__init__.py

```diff
@@ -1163,11 +1145,15 @@ def run(self, root: Element) -> None:
                 continue
 
             if extracted_data:
-                if youtube is not None:
-                    title = self.youtube_title(extracted_data)
-                    if title is not None:
-                        found_url.family.child.text = title
-                    continue
+                if youtube_urls:
```

Comment:
The check for `youtube_urls` seems incorrect here. If a message has two URLs, one of them non-YouTube, we don't want to pass that to the `resize_youtube_image` method.